Assets/Scripts/Common/DamageArea.cs

```diff
@@ -45,23 +45,26 @@ private enum DamageMode {
 			Health health = healthComponent;
 			health.TakeDamage(isEnter ? damage : damage * Time.deltaTime);
 
-			if (!other.attachedRigidbody.CompareTag(effectTarget)) return;
+			if (other.attachedRigidbody.CompareTag(effectTarget))
+				DoDamageEffect(other);
+		}
+	}
 
-			other.GetContacts(contacts);
-			float xValue = 0f;
-			float yValue = 0f;
-			int i = 0;
+	private void DoDamageEffect(Collider2D other) {
+		int contactCount = other.GetContacts(contacts);
+		if (contactCount < 1) return;
 
-			foreach (ContactPoint2D contact in contacts) {
-				if (!contact.collider || contact.collider.gameObject != gameObject) continue;
+		Vector2 averageContact = new Vector2();
+		int num = 0;
 
-				xValue += contact.point.x;
-				yValue += contact.point.y;
-				i++;
-			}
+		foreach (ContactPoint2D contact in contacts) {
+			if (!contact.collider || contact.collider.gameObject != gameObject) continue;
 
-			if (i > 0)
-				onDamageEffect.Invoke(new Vector2(xValue / i, yValue / i));
+			averageContact += contact.point;
+			num++;
 		}
```

Comment:
You're looping through all 10 items in the array regardless of `contactCount`. Old entries in the array will still be considered and can affect the calculation.